Energy-Saver.Tests/UnitTests.cs

```diff
@@ -0,0 +1,213 @@
+        //{
+        //    _logger = logger;
+        //    _suggestionsService = suggestionsService;
+        //    _context = context;
+        //}
+
+            [Fact]
+        public async void TaxInputTest()
+        {
+            using (var db = new EnergySaverTaxesContext(Utilities.TestDbContextOptions()))
+            {
+                var taxes = new Taxes
+                {
+                    ID = 1,
+                    UserID = 4,
+                    Year = 2020,
+                    Month = Months.January,
+                    GasAmount = 1,
+                    ElectricityAmount = 2,
+                    WaterAmount = 3,
+                    HeatingAmount = 4
+                };
+
+                List<Taxes> expectedTaxList = new List<Taxes>();
+                expectedTaxList.Add(taxes);
+
+                db.Taxes.Add(taxes);
+                await db.SaveChangesAsync();
+
+                var actualTaxList = await db.Taxes.ToListAsync();
+
+                Assert.Equal(expectedTaxList, actualTaxList);
+            }
+        }
+
+        [Fact]
+        public async void TaxRemoveTest()
+        {
+            using (var db = new EnergySaverTaxesContext(Utilities.TestDbContextOptions()))
+            {
+                var taxes = new Taxes
+                {
+                    ID = 1,
+                    UserID = 4,
+                    Year = 2020,
+                    Month = Months.January,
+                    GasAmount = 1,
+                    ElectricityAmount = 2,
+                    WaterAmount = 3,
+                    HeatingAmount = 4
+                };
+
+                db.Taxes.Add(taxes);
+                await db.SaveChangesAsync();
+
+                db.Taxes.Remove(taxes);
+                await db.SaveChangesAsync();
+
+                Assert.False(db.Taxes.Any());
+            }
+        }
+
+        [Fact]
+        public async void AscendingOrderListTest()
+        {
+            using (var db = new EnergySaverTaxesContext(Utilities.TestDbContextOptions()))
+            {
+                List<Taxes> actualTaxList = new List<Taxes>();
+                List<Taxes> expectedTaxList = new List<Taxes>();
+
+                var tax1 = new Taxes
+                {
+                    ID = 1,
+                    UserID = 4,
+                    Year = 2020,
+                    Month = Months.January,
+                    GasAmount = 1,
+                    ElectricityAmount = 2,
+                    WaterAmount = 3,
+                    HeatingAmount = 4
+                };
+
+                var tax2 = new Taxes
+                {
+                    ID = 2,
+                    UserID = 4,
+                    Year = 2021,
+                    Month = Months.February,
+                    GasAmount = 2,
+                    ElectricityAmount = 4,
+                    WaterAmount = 3,
+                    HeatingAmount = 2
+                };
```

Comment:
should use an array instead of creating each tax record independently. eg.:
var courses = new Course[]
            {
                new Course{CourseID=1050,Title="Chemistry",Credits=3},
                new Course{CourseID=4022,Title="Microeconomics",Credits=3}
            };

Reply:
There isn't much reason to have an array here. The code becomes slightly longer and it serves no automatic functionality with loops so I don't think there should be an array here